drivers/mtd/mtd_config.c

```diff
@@ -57,15 +57,13 @@
 
 #ifdef CONFIG_MTD_CONFIG_NAMED
 #  define CONFIGDATA_FORMAT_VERSION     1
+#  define MTD_ERASED_ID(dev)            (dev->erasestate)
 #else
 #  define CONFIGDATA_FORMAT_VERSION     2
+#  define MTD_ERASED_ID(dev)            ((dev->erasestate << 8) | dev->erasestate)
```

Comment:
```suggestion
#  define MTD_ERASED_ID(dev)            (((uint16_t)dev->erasestate << 8) | dev->erasestate)
```

Reply:
@pkarashchenko we don't need manually cast since compiler auto promote the smaller type to int or unsigned int.

Reply:
Ok. That is minor comment just to have it explicit like in
```
#  define __swap_uint16(n) \
    (uint16_t)(((((uint16_t)(n)) & 0x00ff) << 8) | \
               ((((uint16_t)(n)) >> 8) & 0x00ff))
```
I'm fine to keep the current version